spring-kafka/src/main/java/org/springframework/kafka/listener/DefaultKafkaConsumerFactory.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.kafka.listener;
+
+import java.util.Properties;
+
+import org.apache.kafka.clients.consumer.Consumer;
+import org.apache.kafka.clients.consumer.KafkaConsumer;
+
+/**
+ * @author Gary Russell
+ *
+ */
+public class DefaultKafkaConsumerFactory<K, V> implements ConsumerFactory<K, V> {
+
+	@Override
+	public Consumer<K, V> createConsumer(Properties configs) {
```

Comment:
We could but it would mean an extra bean in the normal use case (props + factory) instead of just props. i.e. the user will always have to inject a factory; I am not sure what the benefit would be.